views/v_ksm_data_mart.sql

```diff
@@ -581,86 +581,143 @@ Create or Replace View v_datamart_giving as
 
 
 
---- anonymous subquery 
+/* 
+EXCLUSIONS:
+1) Drop all records linked to an entity with the anonymous special handling code
+2) Drop individual gifts with anonymous type 1, 2, 3 for the purpose of calculating the suggested fields�or alternatively drop all such records
 
-With anonymous As (
+*/ 
+
+With a As (
 Select *
 From rpt_pbh634.v_entity_special_handling sh
-Where sh."ANONYMOUS_DONOR" = 'Y'
+--- Drop all records linked to an entity with the anonymous special handling code
+Where sh.ANONYMOUS_DONOR is not null
 ),
 
 hh as (select *
-from rpt_pbh634.v_ksm_giving_trans_hh),
-
-give as (select *
-from RPT_PBH634.v_Ksm_Giving_Summary),
-
-Give_Count as (select hh.ID_NUMBER,
---- Latest Gift
-min (hh.DATE_OF_RECORD) as recent_gift,
-min (hh.FISCAL_YEAR) as recent_gift_fy,
---- Count gifts in this fiscal year
-count (Case When cal.curr_fy = fiscal_year Then hh.fiscal_year Else 0 End) as gift_count_fy,
---- Count gifts in last fiscal year
-count (Case When cal.curr_fy = fiscal_year + 1 Then hh.fiscal_year Else 0 End) as gift_count_pfy1,
---- Count Fiscal Years of Giving 
-count (distinct hh.FISCAL_YEAR) as fy_count,
---- Count of Fiscal Years of Giving (Current FY - Previous 5 FY)
-count(distinct Case When cal.curr_fy = fiscal_year 
- or cal.curr_fy = fiscal_year + 1
- or cal.curr_fy = fiscal_year + 2
- or cal.curr_fy = fiscal_year + 3
- or cal.curr_fy = fiscal_year + 4
- or cal.curr_fy = fiscal_year + 5
- Then hh.FISCAL_YEAR Else 0 End) as giving_last_5_year
+from rpt_pbh634.v_ksm_giving_trans_hh
+--- Drop individual gifts with anonymous type 1, 2, 3 for the purpose of calculating the suggested fields�or alternatively drop all such records
+where trim(rpt_pbh634.v_ksm_giving_trans_hh.ANONYMOUS) is null),
+
+give as (select give.ID_NUMBER, 
+give.af_status,
+rpt_pbh634.ksm_pkg.get_fiscal_year(give.LAST_GIFT_DATE) as last_gift_fy
+from RPT_PBH634.v_Ksm_Giving_Summary give), --- stewardship amount
```

Comment:
This does not follow the anonymous special handling rules used in the hh subquery. Remove v_ksm_giving_summary and use hh instead, e.g.
rpt_pbh634.get_fiscal_year(max(hh.date_of_record)) as last_gift_fy
on line 605 (not tested)